File: app/src/main/java/com/tianchuang/ihome_b/bean/model/PropertyModel.java
package com.tianchuang.ihome_b.bean.model;

import com.tianchuang.ihome_b.bean.PropertyListItemBean;
import com.tianchuang.ihome_b.http.retrofit.HttpModle;
import com.tianchuang.ihome_b.http.retrofit.RetrofitService;

import java.util.ArrayList;

import rx.Observable;

/**
 * Created by Abyss on 2017/2/14.
 * description:物业模块请求网络
 */

public class PropertyModel {
	/**
	 * 物业列表
	 */
	public static Observable<HttpModle<ArrayList<PropertyListItemBean>>> requestPropertyList() {
		return RetrofitService.createShowApi().getPropertyList();
	}

	/**
	 * 设置常用
	 */
	public static Observable<HttpModle<String>> requestSetOften(int id) {
		return RetrofitService.createShowApi().setOften(id);
	}

}

File: app/src/main/java/com/tianchuang/ihome_b/bean/model/PropertyModel.kt
package com.tianchuang.ihome_b.bean.model

import com.tianchuang.ihome_b.bean.PropertyListItemBean
import com.tianchuang.ihome_b.http.retrofit.HttpModle
import com.tianchuang.ihome_b.http.retrofit.RetrofitService

import java.util.ArrayList

import io.reactivex.Observable


/**
 * Created by Abyss on 2017/2/14.
 * description:物业模块请求网络
 */

object PropertyModel {
    /**
     * 物业列表
     */
    fun requestPropertyList(): Observable<HttpModle<ArrayList<PropertyListItemBean>>> {
        return RetrofitService.createShowApi().propertyList
    }

    /**
     * 设置常用
     */
    fun requestSetOften(id: Int): Observable<HttpModle<String>> {
        return RetrofitService.createShowApi().setOften(id)
    }

    /**
     * 物业删除
     */
    fun propertyDelete(id: Int): Observable<HttpModle<String>> {
        return RetrofitService.createShowApi().propertyDelete(id)
    }

}
